e2e_tests/src/lib.rs

```diff
@@ -82,6 +83,11 @@ impl TestClient {
             env_logger::try_init();
         }
 
+        //Check if the environment variable is set, if not use default path
+        if Err(env::VarError::NotPresent) == env::var("PARSEC_SERVICE_ENDPOINT") {
+            env::set_var("PARSEC_SERVICE_ENDPOINT", "unix:/tmp/parsec.sock");
+        }
```

Comment:
I think this approach might be dangerous in terms of threading, see the docs [here](https://doc.rust-lang.org/std/env/fn.set_var.html). Perhaps you can use something like [`static_init`](https://crates.io/crates/static_init) to make sure it only gets called once?

Reply:
Thanks for the comment Ionut. Will fix this :)